avaframe/in4Region/splitInputs.py

```diff
@@ -0,0 +1,296 @@
+"""Helper functions for splitting avalanche input data into individual folders"""
+import logging
+import shapefile  # pyshp
+from shapely.geometry import shape, Point, box, MultiPolygon
+from avaframe.in2Trans import ascUtils
+import pathlib
+
+from avaframe.in3Utils.initializeProject import initializeFolderStruct
+
+# create local logger
+log = logging.getLogger(__name__)
+
+def splitInputsMain(inputDir, outputDir, cfg):
+    """Main function for splitting inputs
+
+    Parameters
+    ----------
+    inputDir: pathlib.Path object
+        path to input directory
+    outputDir: pathlib.Path object
+        path to output directory
+    cfg: dict
+        configuration settings
+
+    Returns
+    -------
+    none
+    """
+    # Fetch the necessary input
+    inputRELDir = inputDir / 'REL'
+    inputShp = next(inputRELDir.glob("*.shp"), None)
+    if not inputShp:
+        log.error(f"No shapefile found in {inputRELDir}.")
+        return
+    inputDEM = next(inputDir.glob("*.asc"), None)
+    if not inputDEM:
+        log.error(f"No DEM file found in {inputDir}.")
+        return
+
+    # Create the output directory
+    outputDir.mkdir(parents=True, exist_ok=True)
+
+    # Step 1: Create the central list
+    log.info("Creating folder list...")
+    folderList = createFolderList(inputShp)
+    # Group folders with identical "name" attributes before the first underscore and update the list
+    folderListGrouped = groupFoldersByName(folderList)
+    log.info("Finished creating folder list")
+
+    # Step 2: Set up ava directories
+    log.info("Running folder initialization for each entry...")
+    for entry in folderListGrouped:
+        folderName = entry['folderName']
+        initializeFolderStruct(str(outputDir / folderName), removeExisting=True)
+        log.debug(f"Created folder structure for '{folderName}'.")
+    log.info("Finished folder initialization")
+
+    # Step 3: Split and move release areas to each directory
+    log.info("Splitting and moving release areas...")
+    splitAndMoveReleaseAreas(folderListGrouped, inputShp, outputDir)
+    log.info("Finished splitting and moving release areas")
+
+    # Step 4: Clip and move DEM
+    if cfg['GENERAL'].getboolean('splitDEM'):
+        log.info("Clipping and moving DEM...")
+        clipDEMByCentroidAndMove(folderListGrouped, inputDEM, outputDir, cfg)
+        log.info("Finished clipping and moving of DEM")
+
+    # (Step 5: Clip and move ENT and RES... other input too?)
+
+    # Step 6: Divide release areas into scenarios based on "scenario" attribute, if it exists
+    log.info("Separating release area by scenario attribute...")
+    splitByScenarios(folderListGrouped, outputDir)
+    log.info("Finished separating by scenarios")
+
+def readShapefile(src):
```

Comment:
I think this should be moved to either in1Data or in2Trans -> shapeUtils.py or similar